lib/translators/json.js

```diff
@@ -158,23 +158,33 @@ function getKeyFilter(keys, ignored) {
   };
   keys = lodash.map(keys, stripSiblingtests);
 
-  reYes = '(' + keys.join('|') + ')';
+  var createTokenizedPattern = function(keys){
+    var re = '(' + keys.join('|') + ')';
 
-  // using a bullet char to avoid conflict with keys with periods
-  reYes = reYes.replace(/\./g, '•');
+    // tokenize escaped dots to avoid confusing them with nested dot notation
+    re = re.replace(/\\\./g, '‣');
```

Comment:
You could use a negative lookback assertion rather than a unicode substitude.

This is just compounding existing hacks. @beck: Have you considered using a different data structure to strings, to represent the JSON path? e.g. an array of keys. I'm guessing it would significantly complicate the filtering though...